build: remove unused REGISTRYURL and GCS download path for registry build

SoumyaRaikwar opened the pull request:
This PR addresses issue #22333.

Changes:
- Drop REGISTRYURL variable and stop propagating it to sub-make.
- Simplify make/photon/Makefile to always build the registry from source via the existing builder script.
- Remove the Google Storage URL that was previously used to download a prebuilt registry binary.

Rationale:
- The REGISTRYURL parameter and the related GCS download path are no longer used in the build pipeline.
- Building the registry from source is the supported path and avoids relying on external binary artifacts.

Notes:
- No Go code changes. CI should verify Makefile behavior remains intact.

MinerYang commented:
Hi @SoumyaRaikwar ,

Thanks for contributing to harbor community.
However the Param is  not unused. It will be consumed on demand.

Best,
Miner

stonezdj commented on make/photon/Makefile:
```diff
@@ -213,10 +208,6 @@ define _extract_archive
 	$(CURL) --connect-timeout 30 -f -k -L $1 | tar xvz -C $2 || exit 1
 endef
 
-define _get_binary
```
keep this function

SoumyaRaikwar replied:
Restored _get_binary. It’s used only by the non-BUILDREG path as requested.

SoumyaRaikwar commented on make/photon/Makefile:
```diff
@@ -174,12 +174,7 @@ _build_nginx:
 	@$(call _build_base,$(REGISTRY),$(DOCKERFILEPATH_REG))
-	@if [ "$(BUILDREG)" != "true" ] ; then \
-		rm -rf $(DOCKERFILEPATH_REG)/binary && mkdir -p $(DOCKERFILEPATH_REG)/binary && \
-		$(call _get_binary, $(REGISTRYURL), $(DOCKERFILEPATH_REG)/binary/registry); \
```
estored the conditional path: when BUILDREG != true, we fetch the registry binary via $(call _get_binary, $(REGISTRYURL), …); otherwise we build from source via the builder script. REGISTRYURL defaults to empty and can be set externally if needed.

SoumyaRaikwar commented:
Thanks for the review. I’ve addressed the feedback:
- Restored REGISTRYURL in the top-level Makefile (left empty by default).
- Passed REGISTRYURL through to the photon sub-make build.
- Restored the conditional download path in make/photon/Makefile for non-BUILDREG builds.
- Restored _get_binary helper.
- Kept removal of the hardcoded GCS URL.

SoumyaRaikwar commented on Makefile:
```diff
@@ -402,12 +402,12 @@ build:
+ -e NPM_REGISTRY=$(NPM_REGISTRY) -e BASEIMAGETAG=$(BASEIMAGETAG) -e IMAGENAMESPACE=$(IMAGENAMESPACE) -e BASEIMAGENAMESPACE=$(BASEIMAGENAMESPACE) \
+ -e REGISTRYURL=$(REGISTRYURL) \
+ -e TRIVY_DOWNLOAD_URL=$(TRIVY_DOWNLOAD_URL) -e TRIVY_ADAPTER_DOWNLOAD_URL=$(TRIVY_ADAPTER_DOWNLOAD_URL) \
+ -e PULL_BASE_FROM_DOCKERHUB=$(PULL_BASE_FROM_DOCKERHUB) -e BUILD_BASE=$(BUILD_BASE) \
```
Thanks for pointing that out. I’ve reverted the indentation to match the existing style

wy65701436 reviewed (approved):
lgtm

SoumyaRaikwar commented:
@stonezdj could you please review my pr

SoumyaRaikwar commented:
@wy65701436 @stonezdj The PR has been approved and auto-merge is enabled, but it seems to be waiting on workflow approval. Could you please approve the CI runs when you get a chance? Thank you!
